pkg/util/common.go

```diff
@@ -1,5 +1,27 @@
 package util
 
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"strings"
+)
+
 const (
 	ExcludedMark = "EXCLUDED"
 )
+
+func SafeAgentConcatName(name ...string) string {
+	fullPath := strings.Join(name, "-")
+	if len(fullPath) < 58 {
+		return fullPath
+	}
+	digest := sha256.Sum256([]byte(fullPath))
+	// since we cut the string in the middle, the last char may not be compatible with what is expected in k8s
+	// we are checking and if necessary removing the last char
+	c := fullPath[50]
```

Comment:
I think this is because when we cut down the long token, we cannot be sure it won't accidentally collide with another name on the system. So, here, we preserve a little more space for incorporating a short and hashed string to make it unique.